### PR TITLE
compile fix for new portmidi versions

### PR DESCRIPTION
**Pull request by ngg**
libporttime.so has been merged into libportmidi.so in later versions.
This fixes compilation on RPM-based systems and on Gentoo.

### CODE REVIEW

**Comment by Jojo-Schmitz** [comment]
Have you signed the [CLA](https://musescore.org(cla) and if so under what name?

**Comment by ngg** [comment]
I've just signed it with the following username: ngg1

File: mscore/CMakeLists.txt [hunk 1]
```diff
@@ -715,7 +715,12 @@ else (MINGW)
            if (APPLE)
                set(PORTMIDI_LIB portmidi)
            else (APPLE)
-               set(PORTMIDI_LIB -lportmidi -lporttime) # Remove -lporttime on RPM-based systems where PortTime is part of PortMidi.
+               find_library(PORTTIME_LIB porttime)
+               if (PORTTIME_LIB)
+                   set(PORTMIDI_LIB portmidi porttime)
```
Comment (mirabilos) [inline mscore/CMakeLists.txt]: Did you forget the `-l` in front of the libraries here?

Comment (ngg) [reply mscore/CMakeLists.txt]: Most of the target_link_libraries() calls in this file doesn't use `-l`, it's more consistent this way (both are correct). It doesn't really matter, but the `if(APPLE)` branch doesn't use `-l` either.

Comment (mirabilos) [reply mscore/CMakeLists.txt]: OK.

**Comment by anatoly-os** [comment]
What is the motivation for this PR? There is neither the related issue nor the description in the PR or commit log.

**Comment by dmitrio95** [comment]
As far as I understand the motivation is that in the newer versions of portmidi there is no `libporttime.so` file anymore which results in MuseScore being not able to build with such versions of this library. At least I understand it so from this PR's description, please correct me if I am wrong.

**Comment by mirabilos** [comment]
Dmitri Ovodok dixit:

>As far as I understand the motivation is that in the newer versions of
>portmidi there is no `libporttime.so` file anymore which results in
>MuseScore being not able to build with such versions of this library.
>At least I understand it so from this PR's description, please correct
>me if I am wrong.

This is the same as I understood. The PR merely makes it so that,
if there’s a libporttime, it’s used together with libportmidi, and
if not, only the latter is used (because it contains the former,
in newer portmidi versions).

**Pull request merged**